power_grid_model_c/power_grid_model_c/src/dataset.cpp

```diff
@@ -31,7 +29,7 @@ PGM_Idx PGM_dataset_info_n_components(PGM_Handle* /*unused*/, PGM_DatasetInfo co
 
 char const* PGM_dataset_info_component_name(PGM_Handle* /*unused*/, PGM_DatasetInfo const* info,
                                             PGM_Idx component_idx) {
-    return info->component_info[component_idx].component->name.c_str();
+    return info->component_info[component_idx].component->name;
```

Comment:
as mentioned before: I would like to maintain the `constexpr string_view` version if possible